app/src/main/java/com/arctouch/codechallenge/listener/InfiniteScrollListener.kt

```diff
@@ -0,0 +1,46 @@
+package com.arctouch.codechallenge.listener
+
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+
+class InfiniteScrollListener(
+        private val linearLayoutManager: LinearLayoutManager,
+        private val listener: OnLoadMoreListener?
+) : RecyclerView.OnScrollListener() {
+
+    private val VISIBLE_THRESHOLD = 2
+    private var loading: Boolean = false // LOAD MORE Progress dialog
+    private var pauseListening = false
+
+
+    private var END_OF_FEED_ADDED = false
+    private val NUM_LOAD_ITEMS = 10
```

Comment:
Why do keep this dead variable/functions  `NUM_LOAD_ITEMS`,`addEndOfRequests` and `pauseScrollListener`?

 What are the consequences of having it?

Reply:
I keep it to use in a likely refactoring later. This isn't good because the project gets disorganized and another developer can use it and cause more bugs.